### DISCUSSION TITLE
Overlapping keymaps on a freshly installed Neovim+LazyVim

### DISCUSSION THREAD

**Discussion by ljfp** [LazyVim/LazyVim#4645] [Q&A] [answered]
I have just installed Neovim (v0.10.2) and LazyVim (v12.43.0) following these instructions: https://www.lazyvim.org/installation

After installing everything I run `:LazyHealth` to see if everything was working as expected. I had a few warnings related to mason.nvim but I didn't care too much about those since they were expected (I don't have PHP, Ruby or Julia installed). However I noticed that there were A LOT of overlapping keymaps! That's definitely something I didn't expect from a fresh installation.

Here are the overlapping keymaps:

----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```
checking for overlapping keymaps ~
- WARNING In mode `n`, <g> overlaps with <gc>, <gco>, <gcO>, <gcc>, <g]>, <g%>, <gx>, <g[>:
  - <g>: goto
  - <gc>: Toggle comment
  - <gco>: Add Comment Below
  - <gcO>: Add Comment Above
  - <gcc>: Toggle comment line
  - <g]>: Move to right "around"
  - <g%>: Cycle backwards through results
  - <gx>: Open with system app
  - <g[>: Move to left "around"
- WARNING In mode `n`, <<Space>w> overlaps with <<Space>wd>, <<Space>wm>, <<C-W><C-D>>, <<C-W><Space>>:
  - <<Space>w>: windows
  - <<Space>wd>: Delete Window
  - <<Space>wm>: Enable Maximize
  - <<C-W><C-D>>: Show diagnostics under the cursor
  - <<C-W><Space>>: Window Hydra Mode (which-key)
- WARNING In mode `x`, <a> overlaps with <al>, <an>, <a%>:
  - <a>: around
  - <al>: last
  - <an>: next
- WARNING In mode `x`, <i> overlaps with <il>, <in>:
  - <i>: inside
  - <il>: last
  - <in>: next
- WARNING In mode `o`, <a> overlaps with <al>, <an>:
  - <a>: around
  - <al>: last
  - <an>: next
- WARNING In mode `o`, <i> overlaps with <il>, <in>:
  - <i>: inside
  - <il>: last
  - <in>: next
- WARNING In mode `n`, <gc> overlaps with <gco>, <gcO>, <gcc>:
  - <gc>: Toggle comment
  - <gco>: Add Comment Below
  - <gcO>: Add Comment Above
  - <gcc>: Toggle comment line
- OK Overlapping keymaps are only reported for informational purposes.
  This doesn't necessarily mean there is a problem with your config.
```

  
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

I'm confused about this. Which ones of those are LazyVim keymaps? For the ones that don't come with LazyVim, where do they came from? Is there any way to easily remove the keymappings that are overlapping with LazyVim defaults?

**Comment by dpetka2001** [comment]
Nothing to worry about. They don't affect your experience. It's just `which-key.nvim` which creates descriptions for groups I believe. It's just the way it works.

**Reply by ljfp** [reply]
Oh, ok. I guess I need to investigate a bit more about the inner workings of `which-key.nvim`. So, just to be sure, there's not actual overlapping of keymaps, right?

**Reply by dpetka2001** [reply, marked as answer]
No, I have the same checkhealth and don't see any problems.
